medusa/providers/generic_provider.py

```diff
@@ -229,6 +229,14 @@ def find_search_results(self, show, episodes, search_mode, forced_search=False,
             return results
 
         if items_list:
+            # Remove duplicate items
+            items_list_without_dups = []
+            for item in items_list:
+                if item['link'] not in [_['link'] for _ in items_list_without_dups]:
+                    items_list_without_dups.append(item)
+
+            items_list = items_list_without_dups
```

Comment:
What is this for and why is it needed now?

Reply:
Because we can't add the same URL in one mass update. And for this feature I can't use the update or insert anymore.

Added benifit is that we don't process/loop through double results.

Reply:
@p0psicles 
```python
unique_items = {item['link']:item for item in itemslist}.values()
```
This uses the `link` as a dictionary key ensuring no duplicates with the original item as the value. The resulting dicts values are now all unique.